tools/c7n_mailer/c7n_mailer/splunk_delivery.py

```diff
@@ -0,0 +1,269 @@
+                        'account', 'unknown'
+                    ),
+                    'sourcetype': '_json',
+                    'index': index,
+                    'event': event
+                })
+        return payloads
+
+    def get_splunk_events(self, msg):
+        """
+        Given a raw c7n message dict, generate a list of per-resource event
+        dictionaries to send to Splunk as events.
+
+        :param msg: c7n notification message
+        :type msg: dict
+        :return: list of Splunk HEC Event dicts
+        :rtype: list
+        """
+        # if an event is present, add it to the log
+        if msg.get('event', None) is not None:
+            # add user to the log message, at the top level...
+            user = get_aws_username_from_event(self.logger, msg['event'])
+            if user is not None:
+                msg['event_triggering_user'] = user
+        # get a copy of the message with no resources
+        base_log = dict(msg)
+        base_log.pop('resources')
+        base_log = deepcopy(base_log)
+        # if configured, build and add actions list
+        if self.config.get('splunk_actions_list', False):
+            base_log['actions'] = []
+            for a in msg['policy']['actions']:
+                if isinstance(a, type({})):
+                    base_log['actions'].append(a['type'])
+                else:
+                    base_log['actions'].append(a)
+        # generate a separate Splunk message for each resource
+        logs = []
+        for res in msg['resources']:
+            x = dict(base_log)
+            if (
+                self.config.get('splunk_remove_metrics', False) and
+                'c7n.metrics' in res
+            ):
+                del res['c7n.metrics']
+            x['resource'] = res
+            # ensure there's one "tags" element, and it's a dict
+            tmp = self.tags_for_resource(res)
+            if 'Tags' in x['resource']:
+                del x['resource']['Tags']
+            x['resource']['tags'] = tmp
+            logs.append(self._prune_log_message(x))
+        return logs
+
+    def _prune_log_message(self, msg):
+        """
+        If the "splunk_remove_paths" config item is not set or empty, return
+        ``msg`` unaltered. Otherwise, for each RFC6901 JSON Pointer in
+        "splunk_remove_paths" that points to an element present in ``msg``,
+        remove that element. Return the result.
+
+        :param msg: Splunk-ready message
+        :type msg: dict
+        :return: msg dict with all ``splunk_remove_paths`` elements removed
+        :rtype: dict
+        """
+        paths = self.config.get('splunk_remove_paths', [])
+        if not paths:
+            return msg
+        for path in paths:
+            try:
+                resolve_pointer(msg, path)
+            except JsonPointerException:
+                continue
+            msg = JsonPatch([{'op': 'remove', 'path': path}]).apply(msg)
```

Comment:
this functionality is interesting, I wonder if its worth exposing more generically, I'm fine with it as is for now. 

one suggestion though would be to build up the patch list, and apply once outside of the loop. the only concern would be overlapped keys, but I think length sorting the remove paths (largest values first), would address that.

Reply:
This would all be much easier if JsonPatch just had an option to not throw an exception if the path doesn't exist...

Ok, I'll push up that change with my next commit.

Reply:
thanks, was just poking at the jsonpatch repo, it looks like a subclass of jsonpatch that overrode the apply method could do try/except JsonPatchError around applying the operations.